server/planning/assignments/assignments_unlink.py

```diff
@@ -28,55 +28,68 @@ def create(self, docs):
-            spike = doc.pop('spike', False)
-
-            # Set the state to 'assigned' if the item is 'submitted'
-            updates = {'assigned_to': deepcopy(assignment.get('assigned_to'))}
-            updates['assigned_to']['state'] = ASSIGNMENT_WORKFLOW_STATE.ASSIGNED
-            assignments_service.patch(assignment[config.ID_FIELD], updates)
-
-            production.system_update(
-                item[config.ID_FIELD],
-                {'assignment_id': None},
-                item
-            )
-
-            # Update published collection
-            if item.get(ITEM_STATE) in PUBLISH_STATES:
-                get_resource_service('published').update_published_items(
+            actioned_item_id = doc.pop('item_id')
+            actioned_item = production.find_one(req=None, _id=actioned_item_id)
+
+            # To support 'unlink as update' we need to link only that update. Not the entire chain
+            if actioned_item and actioned_item.get('rewrite_of'):
+                items = production.find({
+                    'family_id': actioned_item.get('family_id'),
+                    'assignment_id': assignment.get(config.ID_FIELD)
+                })
+            else:
+                items = [actioned_item]
+
+            for item in items:
+                if str(item[config.ID_FIELD]) == str(actioned_item_id):
+                    # Boolean set to true if the unlink is as the result of spiking the content item
+                    spike = doc.pop('spike', False)
+                    doc.update(item)
+
+                    # Set the state to 'assigned' if the item is 'submitted'
+                    updates = {'assigned_to': deepcopy(assignment.get('assigned_to'))}
+                    updates['assigned_to']['state'] = ASSIGNMENT_WORKFLOW_STATE.ASSIGNED
+                    assignments_service.patch(assignment[config.ID_FIELD], updates)
+
+                production.system_update(
                     item[config.ID_FIELD],
-                    'assignment_id', None)
+                    {'assignment_id': None},
+                    item
+                )
+
+                # Update published collection
+                if item.get(ITEM_STATE) in PUBLISH_STATES:
+                    get_resource_service('published').update_published_items(
+                        item[config.ID_FIELD],
+                        'assignment_id', None)
 
-            get_resource_service('delivery').delete_action(lookup={
-                'assignment_id': assignment[config.ID_FIELD],
-                'item_id': item[config.ID_FIELD]
-            })
+                ids.append(item[config.ID_FIELD])
+                updated_items.append(item)
 
-            doc.update(item)
-            ids.append(doc[config.ID_FIELD])
-            items.append(item)
+            # Delete all delivery records for this assignment
+            get_resource_service('delivery').delete_action(lookup={'assignment_id': assignment[config.ID_FIELD]})
```

Comment:
When performing `Unlink as Update` action, won't this remove the delivery records for ALL items?
Should this be removing the delivery records using the `items` array instead?

Reply:
Yup, good catch, will update.